packages/fuels-signers/src/provider.rs

```diff
@@ -60,6 +63,92 @@ impl From<TimeParameters> for FuelTimeParameters {
     }
 }
 
+pub(crate) struct ResourceQueries {
+    utxos: Vec<String>,
+    messages: Vec<String>,
```

Comment:
`ResourceQueries` is hidden from the user and exists only to "encapsulate" the create of the exclusion and spend query. The reason why I don't do it on the fly inside of iterators is because going from `UtxoId` -> `&str` requires you to own the `String` at some point. If it were inside an iterator it's ownership is only temp.

Since this abstraction is used only internally to make our code cleaner, do you think we still need a comment?